apollo-router/src/plugin/mod.rs

```diff
@@ -48,10 +48,33 @@ use crate::RouterResponse;
 use crate::SubgraphRequest;
 use crate::SubgraphResponse;
 
-type InstanceFactory = fn(&serde_json::Value) -> BoxFuture<Result<Box<dyn DynPlugin>, BoxError>>;
+type InstanceFactory =
+    fn(&serde_json::Value, String) -> BoxFuture<Result<Box<dyn DynPlugin>, BoxError>>;
 
 type SchemaFactory = fn(&mut SchemaGenerator) -> schemars::schema::Schema;
 
+/// Initialise details for a plugin
+pub struct PluginInit<T> {
+    /// Configuration
+    pub config: T,
+    /// Router Supergraph Schema
+    pub schema: String,
```

Comment:
Maybe make this non-pub and add an accessor?
That way when apollo-rs is introduced maybe the string schema can just be derived.

Reply:
+1

Also having at least one private field allows us to add more fields later without that being a breaking change

Reply:
I'm going to leave this as is for now and we can consider if we want to change things as part of the 1.0 decision making.